### PR TITLE
Rename to more clear `get_insertion_event_id_by_batch_id` (MSC2716)

### PR DESCRIPTION
**Pull request by MadLittleMods**
Rename to more clear `get_insertion_event_id_by_batch_id`

`get_insertion_event_by_batch_id` -> `get_insertion_event_id_by_batch_id`

Split out from https://github.com/matrix-org/synapse/pull/11114

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] ~~Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)~~
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: changelog.d/11244.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Rename MSC2716 related function to be more clear about the input and output (`get_insertion_event_id_by_batch_id`).
```
Comment (clokep) [inline changelog.d/11244.misc:1]: It might be nice to use the same changelog as #11114 (or maybe reword it as part of that PR) so that towncrier combines them during release.

Comment (MadLittleMods) [reply changelog.d/11244.misc:1]: It seems like the goal is to not burden the changelog with this nothing change. Which might be better suited to just not have a changelog entry for it.

Maybe a new changelog extension that we don't include in the changelog doc. Although `misc` feels pretty relevant for this kind of thing.

I'm kinda leaning towards leaving the changelog entry as-is.

Comment (clokep) [reply changelog.d/11244.misc:1]: > It seems like the goal is to not burden the changelog with this nothing change. Which might be better suited to just not have a changelog entry for it.

This is why I suggested using the same one as #11114, they would then get combined into a single entry.

Comment (MadLittleMods) [reply changelog.d/11244.misc]: Updated with same changelog ⏩

Not confident it staying the same in the other PR.

Comment (clokep) [reply changelog.d/11244.misc]: > Not confident it staying the same in the other PR.

A thing we'll sometimes do is reword *other* changelogs in a PR to combine them.

**Comment by MadLittleMods** [comment]
Thanks for the review @clokep 🦒

File: changelog.d/11244.misc (hunk 1 above)
Comment (MadLittleMods) [reply changelog.d/11244.misc]: Ouu, good thinking! 😁
